Robust.Shared/Serialization/TypeSerializers/Implementations/Custom/TimeOffsetSerializer.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Globalization;
+using JetBrains.Annotations;
+using Robust.Shared.IoC;
+using Robust.Shared.Serialization.Manager;
+using Robust.Shared.Serialization.Markdown;
+using Robust.Shared.Serialization.Markdown.Validation;
+using Robust.Shared.Serialization.Markdown.Value;
+using Robust.Shared.Serialization.TypeSerializers.Interfaces;
+using Robust.Shared.Timing;
+
+namespace Robust.Shared.Serialization.TypeSerializers.Implementations.Custom;
+
+/// <summary>
+/// Offsets the timespan by the CurTime.
+/// </summary>
+public sealed class TimeOffsetSerializer : ITypeSerializer<TimeSpan, ValueDataNode>
+{
+    public TimeSpan Read(ISerializationManager serializationManager, ValueDataNode node,
+        IDependencyCollection dependencies,
+        bool skipHook,
+        ISerializationContext? context = null, TimeSpan value = default)
+    {
+        var seconds = double.Parse(node.Value, CultureInfo.InvariantCulture);
+        var curTime = dependencies.Resolve<IGameTiming>().CurTime.TotalSeconds;
+
+        return TimeSpan.FromSeconds(seconds + curTime);
```

Comment:
Do float conversion before addition to avoid floating point error.

```suggestion
        var curTime = dependencies.Resolve<IGameTiming>().CurTime;

        return curTime + TimeSpan.FromSeconds(seconds);
```